lib/index.js

```diff
@@ -213,36 +218,61 @@ Mongoose.prototype.setDriver = function setDriver(driver) {
  * - 'toJSON': `{ transform: true, flattenDecimals: true }` by default. Overwrites default objects to [`toJSON()`](/docs/api.html#document_Document-toJSON), for determining how Mongoose documents get serialized by `JSON.stringify()`
  * - 'toObject': `{ transform: true, flattenDecimals: true }` by default. Overwrites default objects to [`toObject()`](/docs/api.html#document_Document-toObject)
  *
- * @param {String} key
- * @param {String|Function|Boolean} value
+ * @param {String|Object} key The name of the option or a object of multiple key-value pairs
+ * @param {String|Function|Boolean} value The value of the option, unused if "key" is a object
+ * @returns {Mongoose} The used Mongoose instnace
  * @api public
  */
 
 Mongoose.prototype.set = function(key, value) {
   const _mongoose = this instanceof Mongoose ? this : mongoose;
 
-  if (VALID_OPTIONS.indexOf(key) === -1) {
-    throw new Error(`\`${key}\` is an invalid option.`);
+  if (arguments.length === 1 && typeof key !== 'object') {
+    return _mongoose.options[key];
```

Comment:
Previously we were validating the key even for getters, we should do the same here as well.

Reply:
done with 72bd56cb4022af40413c4f19614c8746d215cac3